libcudacxx/include/cuda/__memory_resource/resource_ref.h

```diff
@@ -45,17 +80,35 @@ enum class _AllocType
 struct _Alloc_vtable
 {
   using _AllocFn   = void* (*) (void*, size_t, size_t);
-  using _DeallocFn = void (*)(void*, void*, size_t, size_t);
+  using _DeallocFn = void (*)(void*, void*, size_t, size_t) noexcept;
   using _EqualFn   = bool (*)(void*, void*);
+  using _DestroyFn = void (*)(void*) noexcept;
+  using _MoveFn    = void (*)(void*, void*) noexcept;
+  using _CopyFn    = void (*)(void*, const void*);
```

Comment:
Question: As stated offline, copying an owning resource might be *really* bad. Do we actually want to allow that in this scenario?

Reply:
the dilemna is that, if a container is storing its memory resource by value, then if the resource isn't copyable, then neither is the container.

Reply:
I thought about this a bit more and I believe we should be able to get around this, bei either allowing the user to pass in a `resource_ref` or create a shared resource type